### PR TITLE
camelCase/PascalCase correction of reflex scaffolding

### LINKED ISSUE #84: scaffolding a reflex produces wrong class name when it contains a _
When creating a reflex using the  management command, it works as long as you don't use a `_` in your reflex name. assuming you have an ap pnamed `core`, start

    ./manage.py generate_reflex core foo_bar

It creates the corresponding files, but according to your javascript it calls the reflex `Foo_BarReflex`

```javascript
    <a
      href="#"
      data-reflex="click->Foo_BarReflex#increment"
      data-count="{{ count }}"
    >Increment </a><span>{{ count }}</span>
```

When you look at the class:

```python
class FooBarReflex(Reflex):
    def increment(self, step=1):
        self.count = int(self.element.dataset['count']) + step
```
Which can't be found. When removing the _ in the `data-reflex` attr, it works.
The generator shouldn't add this underline.

### PR DESCRIPTION
**Pull request by nerdoc**
fixes #84

### CODE REVIEW

**Comment by nerdoc** [comment]
Again... test suite locally ran without problems. please fix your CI... and use black ;-)

**Comment by nerdoc** [comment]
Please wait merging this PR. I found another issue - the reflexes are named correctly now, but the other way round seems to be another problem: HTML: `data-reflex="click->Foo_Reflex#foo"`, then error: 

    'servermessage': {...  "SockpuppetConsumer failed to invoke Foo_Reflex#foo, with url http://127.0.0.1:8000/bar/. AttributeError: 'FooReflex' object has no attribute 'foo'" ...}

So once the (manually written) wrong reflex name is reversed correctly into *FooReflex*, but it tries to invoke *Foo_Reflex*. Need to investigate further.

File: .gitignore [hunk 1]
```diff
@@ -67,3 +67,4 @@ jsdist/
 cypress/videos/
 cypress/screenshots/
 slides
+sockpuppet/static/
```
Comment (jonathan-s) [inline .gitignore:70]: This can't be ignored, as these need to be committed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## [Unreleased](https://github.com/jonathan-s/django-sockpuppet/tree/HEAD)
 
+- improved tests documentation
+- fixed camelcase/pascalcase tools for Reflex names
```
Comment (jonathan-s) [inline CHANGELOG.md:6]: Changelog is actually automatically generated so this is not needed :). But thanks for thinking of it.

**Comment by jonathan-s** [comment]
@nerdoc I'm afraid that all of the integration tests fail so this is a no go at the moment. If you make sure that they pass I think this looks promising.

File: .gitignore (hunk 1 above)
Comment (nerdoc) [reply .gitignore:70]: ok!

File: CHANGELOG.md (hunk 2 above)
Comment (nerdoc) [reply CHANGELOG.md:6]: ah - I thought that - do you use a tool here? looks quite interesting...

Comment (jonathan-s) [reply CHANGELOG.md:6]: You can find the config of it here: https://github.com/jonathan-s/django-sockpuppet/blob/master/.github/workflows/changelog.yml

File: tests/test_filters.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    def test_preserve_already_pascalcase(self):
+        self.assertEqual(pascalcase("FooBar"), "FooBar")
+
+    def test_preserve_already_3_pascalcase(self):
+        self.assertEqual(pascalcase("FooBarBaz"), "FooBarBaz")
+
+    def test_camelcase2pascalcase(self):
+        self.assertEqual(pascalcase("fooBarBaz"), "FooBarBaz")
+
+    # def test_preserve_CAPITALS_pascalcase(self):
```
Comment (jonathan-s) [inline tests/test_filters.py:60]: Is there a reason for this being commented?

File: tests/test_filters.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    def test_preserve_already_camelcase(self):
+        self.assertEqual(camelcase("foo"), "foo")
+
+    def test_preserve_already_3_camelcase(self):
+        self.assertEqual(camelcase("fooBarBaz"), "fooBarBaz")
+
+    def test_pascalcase2camelcase(self):
+        self.assertEqual(camelcase("FooBarBaz"), "fooBarBaz")
+
+    # def test_preserve_CAPITALS_camelcase(self):
```
Comment (jonathan-s) [inline tests/test_filters.py:25]: Same question here?

File: tests/test_filters.py (hunk 3 above)
Comment (nerdoc) [reply tests/test_filters.py:60]: ah - i just wanted you to know that I considered leaving the possibility of UPPERCASE blocks in the code - but this is such a small user case. Maybe it's better to just remove it, as the camelcase Js counterpart of stimulus-reflex is not able to do that.
I'll remove that.

File: tests/test_filters.py (hunk 4 above)
Comment (nerdoc) [reply tests/test_filters.py]: done

**Comment by jonathan-s** [comment]
Thanks for making this and apologies for the long delay!

**Comment by nerdoc** [comment]
no problem - I'm glad I could help a bit... ATM I switched to unicorn as Sockpuppet had to many quirks to setup. But websockets is the better alternative on the long term, so... I'll keep an eye on it :smiley:

**Pull request merged**